cmd/service_bootstrap_desktop.go

```diff
@@ -49,27 +51,79 @@ func (di *Dependencies) BootstrapServices(nodeOptions node.Options) error {
 	return nil
 }
 
+const logPrefix = "[service bootstrap] "
+
+type locationInfo struct {
+	OutIP   string
+	PubIP   string
+	Country string
+}
+
+func (di *Dependencies) resolveIPsAndLocation() (loc locationInfo, err error) {
+	pubIP, err := di.IPResolver.GetPublicIP()
+	if err != nil {
+		return
+	}
+	loc.PubIP = pubIP
+
+	outboundIP, err := di.IPResolver.GetOutboundIP()
+	if err != nil {
+		return
+	}
+	loc.OutIP = outboundIP
+
+	currentCountry, err := di.LocationResolver.ResolveCountry(pubIP)
+	if err != nil {
+		log.Warn(logPrefix, "Failed to detect service country. ", err)
+		err = service.ErrorLocation
+		return
+	}
+	loc.Country = currentCountry
+
+	log.Info(logPrefix, "Detected service country: ", loc.Country)
+
+	return
+}
+
 func (di *Dependencies) bootstrapServiceOpenvpn(nodeOptions node.Options) {
-	createService := func(serviceOptions service.Options) (service.Service, error) {
+	createService := func(serviceOptions service.Options) (service.Service, market.ServiceProposal, error) {
+		location, err := di.resolveIPsAndLocation()
```

Comment:
I imagine that things like (location, ip address, etc.) should be asked from `nodeContext`, instead if fetching it twice

Reply:
We don't have a component for this though, there's a location cache, which is rather strange but nothing like that for ip resolution. Created #653 for this. Out of scope for this PR.

Reply:
I am saying that this duplication indicates that its tine to refactor towards better arcitecture